modules/claims_api/app/services/claims_api/dependent_claimant_poa_assignment_service.rb

```
@@ -0,0 +1,169 @@
+        log(detail: 'POA assigned to dependent.')
+
+        return true
+      end
+
+      false
+    end
+
+    def dependent_claims
+      local_bgs = ClaimsApi::LocalBGS.new(external_uid: @dependent_participant_id,
+                                          external_key: @dependent_participant_id)
+      res = local_bgs.find_benefit_claims_status_by_ptcpnt_id(@dependent_participant_id)
+
+      return res&.dig(:benefit_claims_dto, :benefit_claim) if res&.dig(:benefit_claims_dto, :benefit_claim).present?
+
+      log(level: :error, detail: 'Dependent claims not found in BGS')
+
+      raise ::Common::Exceptions::FailedDependency
+    end
+
+    def benefit_claim_web_service
+      ClaimsApi::BenefitClaimWebService.new(external_uid: @dependent_participant_id,
+                                            external_key: @dependent_participant_id)
+    end
+
+    def benefit_claim_service
+      ClaimsApi::BenefitClaimService.new(external_uid: @dependent_participant_id,
+                                         external_key: @dependent_participant_id)
+    end
+
+    def claim_details(claim_id)
+      res = benefit_claim_web_service.find_bnft_claim(claim_id:)
+
+      return res&.dig(:bnft_claim_dto) if res&.dig(:bnft_claim_dto).present?
+
+      log(level: :error, detail: 'Claim details not found in BGS', claim_id:)
+
+      raise ::Common::Exceptions::FailedDependency
+    end
+
+    def poa_participant_id
+      poa_ptcpnt = FindPOAsService.new.response.find { |combo| combo[:legacy_poa_cd] == @poa_code }
+
+      return poa_ptcpnt&.dig(:ptcpnt_id) if poa_ptcpnt&.dig(:ptcpnt_id).present?
+
+      log(level: :error, detail: 'POA code/participant ID combo not found in BGS')
+
+      raise ::Common::Exceptions::FailedDependency
+    end
+
+    def manage_ptcpnt_rlnshp_poa_success?(response)
+      response.is_a?(Hash) && response.dig(:comp_id, :ptcpnt_rlnshp_type_nm) == 'Power of Attorney For'
```

Brittle check. Any better suggestions?

---

You could add some error handling that would alert you if the structure changed. 

Another idea is something like this:
```ruby
response.is_a?(Hash) && response.dig(:comp_id, :ptcpnt_rlnshp_type_nm)&.casecmp?('Power of Attorney For')
```
chatgpt suggested that, so definitely double check it's doing what you want 😃

---

I thought about normalizing the strings, but I think failing loudly is our best bet at fixing the issue. I'm not sure trying to anticipate changing punctuation or cases is worthwhile 🤷